Spanner/tests/System/TransactionTest.php

```diff
@@ -264,6 +265,112 @@ public function testRunTransactionWithDbRole($db, $values, $expected)
         }
     }
 
+    // @TODO: Need to work on this once the directedRead is available
+    /**
+     * @dataProvider getDirectedReadOptions
+     */
+    public function testRWTransactionExecuteWithDirectedReadOptions($directedReadOptions)
+    {
+        $this->markTestSkipped('fixme');
+        $db = self::$database;
+        $transaction = $db->transaction();
+        $expected = 'Directed reads can only be performed in a read-only transaction.';
+        $exception = null;
+
+        try {
+            $row = $db->execute(
+                'SELECT * FROM ' . self::$tableName,
+                ['transactionId' => $transaction->id()] + $directedReadOptions
+            )->rows()->current();
+        } catch (ServiceException $e) {
+            $exception = $e;
+        }
+        $this->assertEquals($exception->getReason(), $expected);
+        $exception = null;
+
+        try {
+            $row = $transaction->execute(
+                'SELECT * FROM ' . self::$tableName,
+                $directedReadOptions
+            )->rows()->current();
+        } catch (ServiceException $e) {
+            $exception = $e;
+        }
+        $this->assertEquals($exception->getReason(), $expected);
```

Comment:
I don't understand why this is repeated

Reply:
Please see that, one is `$db->execute()` and the other one is `transaction->execute()`. Thats the difference here. I am trying to use functions from both the classes to see they are working properly!

Reply:
understood, tnx